pkg/asset/cluster/tfvars/tfvars.go

```diff
@@ -485,6 +485,18 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		if err != nil {
 			return err
 		}
+		// Based on the number of workers, we could have the following outcomes:
+		// 1. workers > 0, masters not schedulable, valid cluster
+		// 2. workers = 0, masters schedulable, valid compact cluster but currently unsupported on GCP
+		// 3. workers = 0, masters not schedulable, invalid cluster
+		if len(workers) == 0 {
+			switch {
+			case mastersSchedulable:
+				return errors.Errorf("0 workers with masters being schedulable for workloads is currently unsupported on this platform")
+			default:
+				return errors.Errorf("cannot have 0 workers when masters are not schedulable")
+			}
+		}
```

Comment:
So we cannot check this during install-config validation because this happens after the manifests have been generated and (possibly) edited by users.
If this becomes a pattern, we should consider adding a manifest validation stage.